### PR TITLE
Fix 3744

### PR DESCRIPTION
**Pull request by nakai-omer**
Add better explanation on how to integrate collision monitor with nav2 stack.

### CODE REVIEW

File: configuration/packages/configuring-collision-monitor.rst [hunk 1]
```diff
@@ -75,7 +75,7 @@ Parameters
   ============== =============================
   Type           Default
   -------------- -----------------------------
-  string         "cmd_vel_raw"
+  string         "cmd_vel_smoothed"
```
Comment (SteveMacenski) [inline configuration/packages/configuring-collision-monitor.rst:78]: Make a note in the description that pre-`Jazzy` this was set to `cmd_vel_raw` by default

File: tutorials/docs/using_collision_monitor.rst [hunk 2]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
 ====================
 
 The Collision Monitor is designed to operate below Nav2 as an independent safety node.
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
+This acts as a filter on the ``cmd_vel`` topic coming out of the Velocity Smoother Server.
```
Comment (SteveMacenski) [inline tutorials/docs/using_collision_monitor.rst:114]: Revert. This is all still acting as a collision filter for the controller, which the velocity smoother knows nothing about. 

It might be better to be reworded that its filtering the cmd_vel from the controller for potential collisions (and less "of the topic coming out of...") to be abstract from the actual chain ordering. The velocity smoother, while present, its really part of the concept that is needed to  be explained to understand and use this node

File: tutorials/docs/using_collision_monitor.rst [hunk 3]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
...
+If no such zone is triggered, then the Velocity Smoother's ``cmd_vel`` message is used.
 Else, it is scaled or set to stop as appropriate.
-For correct operation of the Collision Monitor with the Controller, it is required to add the ``cmd_vel -> cmd_vel_raw`` remapping to the ``navigation_launch.py`` bringup script as presented below:
+For correct operation of the Collision Monitor with the Velocity Smoother, it is required to remove the Velocity Smoother's `cmd_vel_smoothed` remapping in the ``navigation_launch.py`` bringup script as presented below:
```
Comment (SteveMacenski) [inline tutorials/docs/using_collision_monitor.rst:117]:
```suggestion
For correct operation of the Collision Monitor with the Velocity Smoother, it is required to remove the Velocity Smoother's `cmd_vel_smoothed` remapping in the ``navigation_launch.py`` bringup script as presented below to make the output of the server ``cmd_vel_smoothed``, which will be the input to the newly added collision monitor:
```

File: tutorials/docs/using_collision_monitor.rst [hunk 4]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
...
         parameters=[configured_params],
-    +   remappings=remappings + [('cmd_vel', 'cmd_vel_raw')]),
+        remappings=remappings +
+    -              [('cmd_vel', 'cmd_vel_nav'), ('cmd_vel_smoothed', 'cmd_vel')]),
+    +              [('cmd_vel', 'cmd_vel_nav')]),
 
-Please note, that the remapped ``cmd_vel_raw`` topic should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement.
+If you have changed Collision Monitor's default ``cmd_vel_in_topic`` and ``cmd_vel_out_topic`` configuration, make sure Velocity Smoother's default output topic ``cmd_vel_smoothed`` should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement. 
 
 Demo Execution
```
Comment (SteveMacenski) [inline tutorials/docs/using_collision_monitor.rst:145]: @AlexeyMerzlyakov one thing I note about this tutorial is that there's no integration of this info into the core navigation launch file if someone wants to use this as a component in navigation for their system (only uses a separate launch file). Should we add that step here to integrate this into the core navigation launch file?

File: configuration/packages/configuring-collision-monitor.rst (hunk 1 above)
Comment (nakai-omer) [reply configuration/packages/configuring-collision-monitor.rst:78]: Done

File: tutorials/docs/using_collision_monitor.rst (hunk 2 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst:114]: I have reworded it. Just not so sure about the last sentence:
> its really part of the concept that is needed to be explained to understand and use this node
Do you want me to add an explanation about the velocity smoother?

File: tutorials/docs/using_collision_monitor.rst (hunk 3 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: Done

File: tutorials/docs/using_collision_monitor.rst [hunk 5]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
 ====================
 
 The Collision Monitor is designed to operate below Nav2 as an independent safety node.
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
+It acts as a filter for the `cmd_vel` messages from the controller to avoid potential collisions.
+If no such zone is triggered, then the Velocity Smoother's ``cmd_vel`` message is used.
 Else, it is scaled or set to stop as appropriate.
-For correct operation of the Collision Monitor with the Controller, it is required to add the ``cmd_vel -> cmd_vel_raw`` remapping to the ``navigation_launch.py`` bringup script as presented below:
+For correct operation of the Collision Monitor with the Velocity Smoother, it is required to remove the Velocity Smoother's `cmd_vel_smoothed` remapping in the ``navigation_launch.py`` bringup script as presented below to make the output topic of the server ``cmd_vel_smoothed``, which will be the input to the newly added collision monitor:
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:117]: I think, here should be added one more "Note" section, to tell that despite we are using `cmd_vel_smoothed`, going out of Smoother, this is not essential for Collision Monitor. CM could use any input velocity going out from any chain of developer's configuration. It could be output non-smoothed velocity from Controller Server, e.g. if Velocity Smoother is not important in the system, or anything else.

The most important message here is that CM should act as safety node, and thus to be the last chain in velocities (post)processing.

**Review by AlexeyMerzlyakov — COMMENTED** [review]
Mostly, I agree with the idea and update. Please, check the review comments below.

File: tutorials/docs/using_collision_monitor.rst [hunk 6]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
 ====================
 
 The Collision Monitor is designed to operate below Nav2 as an independent safety node.
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
+It acts as a filter for the `cmd_vel` messages from the controller to avoid potential collisions.
+If no such zone is triggered, then the Velocity Smoother's ``cmd_vel`` message is used.
```
Comment (SteveMacenski) [inline tutorials/docs/using_collision_monitor.rst:115]: Remove velcoity smoother.

File: tutorials/docs/using_collision_monitor.rst [hunk 7]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
 ====================
 
 The Collision Monitor is designed to operate below Nav2 as an independent safety node.
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
+It acts as a filter for the `cmd_vel` messages from the controller to avoid potential collisions.
```
[the remark this answers is not in the document]
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: done

File: tutorials/docs/using_collision_monitor.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: done

File: tutorials/docs/using_collision_monitor.rst [hunk 8]
```diff
@@ -111,35 +111,41 @@ Preparing Nav2 stack
...
+        plugin='nav2_velocity_smoother::VelocitySmoother',
+        name='velocity_smoother',
         parameters=[configured_params],
-    +   remappings=remappings + [('cmd_vel', 'cmd_vel_raw')]),
+        remappings=remappings +
+    -              [('cmd_vel', 'cmd_vel_nav'), ('cmd_vel_smoothed', 'cmd_vel')]),
+    +              [('cmd_vel', 'cmd_vel_nav')]),
 
-Please note, that the remapped ``cmd_vel_raw`` topic should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement.
+If you have changed Collision Monitor's default ``cmd_vel_in_topic`` and ``cmd_vel_out_topic`` configuration, make sure Velocity Smoother's default output topic ``cmd_vel_smoothed`` should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement.
```
[the remark this answers is not in the document]
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: removed

File: tutorials/docs/using_collision_monitor.rst (hunk 5 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: I have split it into two subsections, one for nav2 bringup, and one for custom bringup scripts. lmkwyt please.

File: tutorials/docs/using_collision_monitor.rst (hunk 6 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: removed

File: tutorials/docs/using_collision_monitor.rst [hunk 9]
```diff
@@ -111,35 +111,48 @@ Preparing Nav2 stack
...
 The Collision Monitor is designed to operate below Nav2 as an independent safety node.
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
-Else, it is scaled or set to stop as appropriate.
-For correct operation of the Collision Monitor with the Controller, it is required to add the ``cmd_vel -> cmd_vel_raw`` remapping to the ``navigation_launch.py`` bringup script as presented below:
+It acts as a filter for the ``cmd_vel`` messages from the controller to avoid potential collisions.
+If no such zone is triggered, then the ``cmd_vel`` message is used.
+Else, it is scaled or set to stop as appropriate. 
+
+Using with NAV2 Bringup
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:118]: Here and in all places: NAV2 -> Nav2

File: tutorials/docs/using_collision_monitor.rst [hunk 10]
```diff
@@ -111,35 +111,48 @@ Preparing Nav2 stack
...
+-----------------------
+By default, the Collision Monitor is configured for usage with the NAV2 bringup package, using the ``navigation_launch.py`` launch file. For correct operation of the Collision Monitor with the Velocity Smoother, it is required to remove the Velocity Smoother's ``cmd_vel_smoothed`` remapping in the ``navigation_launch.py`` bringup script as presented below to make the output topic of the server equal ``cmd_vel_smoothed``, which will be the input to the newly added collision monitor:
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:120]: Change to something like:
```
For correct operation of the Collision Monitor with the Velocity Smoother, it is required to remove the Velocity Smoother's ``cmd_vel_smoothed`` remapping in the ``navigation_launch.py`` bringup script as presented below. This will make the output topic of the Veclory Smoother to be untouched, which will be the input to the newly added Collision Monitor:
```

File: tutorials/docs/using_collision_monitor.rst [hunk 11]
```diff
@@ -111,35 +111,48 @@ Preparing Nav2 stack
...
-    +   remappings=remappings + [('cmd_vel', 'cmd_vel_raw')]),
+        remappings=remappings +
+    -              [('cmd_vel', 'cmd_vel_nav'), ('cmd_vel_smoothed', 'cmd_vel')]),
+    +              [('cmd_vel', 'cmd_vel_nav')]),
+
+If you have changed Collision Monitor's default ``cmd_vel_in_topic`` and ``cmd_vel_out_topic`` configuration, make sure Velocity Smoother's default output topic ``cmd_vel_smoothed`` should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement.
 
-Please note, that the remapped ``cmd_vel_raw`` topic should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement.
+Using with a custom launch file
+-------------------------------
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:149]: I think, this is better to be changed to `.. note::` insertion. This is definitely is out of scope of this tutorial, so this should not be a section, everyone will read to do not understand whether they should enable CM in `navigation_bringup.launch` or somewhere else.

File: tutorials/docs/using_collision_monitor.rst [hunk 12]
```diff
@@ -111,35 +111,48 @@ Preparing Nav2 stack
...
+As the Collision Monitor acts as a safety node, it must be the last link in the velocity message chain, making it the node that publishs to the ``cmd_vel`` topic. Therefore, in any custom NAV2 launch configuration, the current node publishing to the ``cmd_vel`` topic, should be remapped to publish to the topic configured in Collision Montior's ``cmd_vel_in_topic`` configuration (``cmd_vel_smoothed`` by default).
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:150]: velocity message chain -> velocity message post-processing chain

Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:150]: Please insert some clarification after first sentence. Something, like:

It could be placed after smoothed velocity, as in our demonstration, or after non-smoothed velocity from Controller Server, e.g. if Velocity Smoother was not enabled in the system, or going after any other module producing the end-velocity.

Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:150]: * the current node publishing ... -> the end-node publishing ...
* should be remapped to publish to the topic configured in Collision Montior's ``cmd_vel_in_topic`` configuration -> should be remapped to publish to the Collision Monitor input topic configured ``cmd_vel_in_topic`` ROS-parameter

File: tutorials/docs/using_collision_monitor.rst [hunk 13]
```diff
@@ -111,35 +111,48 @@ Preparing Nav2 stack
...
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
-Else, it is scaled or set to stop as appropriate.
-For correct operation of the Collision Monitor with the Controller, it is required to add the ``cmd_vel -> cmd_vel_raw`` remapping to the ``navigation_launch.py`` bringup script as presented below:
+It acts as a filter for the ``cmd_vel`` messages from the controller to avoid potential collisions.
+If no such zone is triggered, then the ``cmd_vel`` message is used.
+Else, it is scaled or set to stop as appropriate. 
+
+Using with NAV2 Bringup
+-----------------------
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:119]: I think, after all, "Using with Nav2 Bringup" sub-section is not needed. We are just using bringup configuration by-default, and just telling the developer about other abilities in the notice (to not make someone be confused that this note text is also something, he need to to)

File: tutorials/docs/using_collision_monitor.rst (hunk 5 above)
Comment (AlexeyMerzlyakov) [reply tutorials/docs/using_collision_monitor.rst]: Good re-work. However, I still consider that "note" section is better, since we won't confuse the developer who reads this tutorial by unnecessary actions (please read the comments below from next review round for better understanding what I mean).

File: tutorials/docs/using_collision_monitor.rst (hunk 9 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst:118]: Done

File: tutorials/docs/using_collision_monitor.rst (hunk 12 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst:150]: done

Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst:150]: done

Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst:150]: done

File: tutorials/docs/using_collision_monitor.rst (hunk 13 above)
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst:119]: Removed

**Comment by nakai-omer** [comment]
@AlexeyMerzlyakov Thanks for investing so much time in this PR, I have updated the code to meet your requests.

File: tutorials/docs/using_collision_monitor.rst [hunk 14]
```diff
@@ -111,35 +111,44 @@ Preparing Nav2 stack
...
-This acts as a filter on the ``cmd_vel`` topic coming out of the Controller Server.
-If no such zone is triggered, then the Controller's ``cmd_vel`` is used.
-Else, it is scaled or set to stop as appropriate.
-For correct operation of the Collision Monitor with the Controller, it is required to add the ``cmd_vel -> cmd_vel_raw`` remapping to the ``navigation_launch.py`` bringup script as presented below:
+It acts as a filter for the ``cmd_vel`` messages from the controller to avoid potential collisions.
+If no such zone is triggered, then the ``cmd_vel`` message is used.
+Else, it is scaled or set to stop as appropriate. 
+
+By default, the Collision Monitor is configured for usage with the Nav2 bringup package, using the ``navigation_launch.py`` launch file. For correct operation of the Collision Monitor with the Velocity Smoother, it is required to remove the Velocity Smoother's ``cmd_vel_smoothed`` remapping in the ``navigation_launch.py`` bringup script as presented below. This will make the output topic of the Velocity Smoother to be untouched, which will be the input to the newly added Collision Monitor:
```
Comment (AlexeyMerzlyakov) [inline tutorials/docs/using_collision_monitor.rst:118]: ... using the ``navigation_launch.py`` launch file.
->
... running in parallel with the ``navigation_launch.py`` launch file

Otherwise, it might be implicitly thought that CM launch-file is running `navigation_launch.py` inside which is not true.

**Review by AlexeyMerzlyakov — COMMENTED** [review]
A few nitpick comments and I think, this will be ready ToGo

File: tutorials/docs/using_collision_monitor.rst [hunk 15]
```diff
@@ -111,35 +111,44 @@ Preparing Nav2 stack
...
 
-Please note, that the remapped ``cmd_vel_raw`` topic should match to the input velocity ``cmd_vel_in_topic`` parameter value of the Collision Monitor node, and the output velocity ``cmd_vel_out_topic`` parameter value should be actual ``cmd_vel`` to fit the replacement.
+.. note:: As the Collision Monitor acts as a safety node, it must be the last link in the velocity message post-processing chain, making it the node that publishes to the ``cmd_vel`` topic. It could be placed after smoothed velocity, as in our demonstration, or after non-smoothed velocity from Controller Server, e.g. if Velocity Smoother was not enabled in the system, or going after any other module in custom configuration producing the end-velocity. Therefore, in any custom Nav2 launch configuration, the last node publishing to the ``cmd_vel`` topic, should be remapped to publish to the Collision Monitor input topic configured by cmd_vel_in_topic ROS-parameter (``cmd_vel_smoothed`` by default).
```
[the remark this answers is not in the document]
Comment (nakai-omer) [reply tutorials/docs/using_collision_monitor.rst]: done

File: configuration/packages/collision_monitor/configuring-collision-monitor-node.rst [hunk 16]
```diff
@@ -75,11 +75,12 @@ Parameters
   ============== =============================
   Type           Default
   -------------- -----------------------------
-  string         "cmd_vel_raw"
+  string         "cmd_vel_smoothed"
   ============== =============================
 
   Description:
     Input ``cmd_vel`` topic with desired robot velocity.
+    Please note, pre-`Jazzy` this was set to `cmd_vel_raw` by default.
```
Comment (AlexeyMerzlyakov) [inline configuration/packages/collision_monitor/configuring-collision-monitor-node.rst:83]:
```suggestion
    Please note, pre-`Jazzy` this was set to ``cmd_vel_raw`` by default.
```

**Review by AlexeyMerzlyakov — APPROVED** [review]
One more change and we could merge it to master

File: configuration/packages/collision_monitor/configuring-collision-monitor-node.rst [hunk 17]
```diff
@@ -75,11 +75,12 @@ Parameters
   ============== =============================
   Type           Default
   -------------- -----------------------------
-  string         "cmd_vel_raw"
+  string         "cmd_vel_smoothed"
   ============== =============================
 
   Description:
     Input ``cmd_vel`` topic with desired robot velocity.
+    Please note, pre-`Jazzy` this was set to ``cmd_vel_raw`` by default.
```
Comment (AlexeyMerzlyakov) [inline configuration/packages/collision_monitor/configuring-collision-monitor-node.rst:83]:
```suggestion
    Please note, pre-``Jazzy`` this was set to ``cmd_vel_raw`` by default.
```

**Comment by AlexeyMerzlyakov** [comment]
Merged to master, please sync-up changes from this PR with `README.md` in https://github.com/ros-planning/navigation2/pull/3745
